lib/exclusive-lock.js

```diff
@@ -0,0 +1,157 @@
+'use strict'
+
+const Ajv = require('ajv')
+const schemas = require('./schemas/index.js')
+const jsonParseFilter = require('./json-parse-filter.js')
+
+const DEFAULT_LOCK_TTL_MS = 3000
+const DEFAULT_LOCK_REFRESH_MS = 1000
+const REQUIRED_MIN_MS_DIFF = 500
+
+const ajv = new Ajv()
+const validate = ajv.compile(schemas.instantiation_options)
+
+class ExclusiveLock {
+  #lock_name
+
+  constructor(opts) {
+    opts = opts || {}
+
+    if (!opts.log?.error) {
+      const err = new Error('A pino logger instance is required')
+      err.code = 'EINVAL'
+      throw err
+    }
+
+    this.log = opts.log
+
+    if (!validate(opts)) {
+      const err = new Error('Input validation failed')
+      err.code = 'EINVAL'
+      err.errors = validate.errors.map((e) => {
+        const {instancePath, message} = e
+        return {instancePath, message}
+      })
+      this.log.error(err)
+      throw err
+    }
+
+    const {
+      app_name
+    , log
+    , cache_connection
+    , lock_ttl_ms = DEFAULT_LOCK_TTL_MS
+    , lock_refresh_ms = DEFAULT_LOCK_REFRESH_MS
+    , lock_contents = 1 // Can be specified for more context, e.g. POD_NAME
```

Comment:
is this just a label of some sort?

Reply:
That's that actual content stored in the record by default. Just something minimal and truthy.  It can be overridden with lock_contents